subsys/jwt/jwt.h

```diff
@@ -0,0 +1,8 @@
+/*
+ * Copyright (C) 2024 BayLibre SAS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+int jwt_sign_impl(struct jwt_builder *builder, const unsigned char *der_key,
```

Comment:
you may want some header guards here.  Additionally for this to be stand-alone the include where `jwt_builder` is defined should be added to the file... or forward declared as it's a pointer to force the compiler to warn about transitive includes.

Reply:
You're absolutely right. Thanks for both suggestions!